### ISSUE TITLE
Jetpack Focus: Enable app extensions for Jetpack

### ISSUE THREAD

**Issue by hassaanelgarem** [wordpress-mobile/WordPress-iOS#19361]
## Description

We have quite a few app extensions for the WordPress app. We want to replicate them for the Jetpack app as well 
Asana task: 1202650603324029-as-1203060871421556/f

## Tasks

- [x] Add the necessary provisioning profiles (#19384)
- [x] Share Extension (#19383)
- [x] Draft Action Extension (#19414)
- [x] ~~Notification Content Extension~~ (Descoped. Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/19361#issuecomment-1271060287)
- [x] Notification Service Extension (#19415)
- [x] ~~Siri Intents~~ (Descoped. Ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/19361#issuecomment-1275471543)
- [x] Update Draft action copy. (Ref: p1664934688068699-slack-C0180B5PRJ4) (#19456)

**Comment by hassaanelgarem** [comment]
Notification content extension turned out to not be currently used by WP. We depend solely on the service extension. So we won't be adding for Jetpack.

**Comment by hassaanelgarem** [comment]
`WordPressIntents` has a dependency on the WordPress widgets. Because of this, adding Siri intents for Jetpack will be done within the Jetpack Widgets project. (Ref: 1202650603324029-as-1202700379456805/f)

**Comment by staskus** [comment]
@hassaanelgarem 

When moving Jetpack Widgets, I noticed that there're constants in `Constants.m` that are shared between Jetpack and WordPress apps. See `Notification Content Extension Constants`, `Notification Service Extension Constants`, and `Share Extension Constants`. 

These constants are used as keys for shared `UserDefaults`, for example, `userDefaults.set(defaultSiteID, forKey: WPShareExtensionUserDefaultsPrimarySiteID)`.

With Shared Login project we started sharing [same app group between WordPress and Jetpack apps](https://github.com/wordpress-mobile/WordPress-iOS/pull/19328). It means that if we set a value in shared `UserDefaults`, it would be reached by both `Jetpack` and `WordPress` app. Therefore, having same keys in `Contants.m` could cause some issues.

I haven't analysed if there're any issues caused by using same keys for Jetpack and WordPress extensions, but I'm letting you know. It may be worth taking a look.
